### PR TITLE
E2E test for ckBTC

### PR DESCRIPTION
**Pull request by dskloetd**
# Motivation

We don't have an e2e test yet for ckBTC.
I recently set up the test environment such that it's possible to mock incoming BTC transactions by clicking "Get BTC" (where normally the "Get ICP button is).

# Changes

1. Enable `ENABLE_CKBTC` on local.
2. Add necessary test IDs.
3. Add page objects.
4. Add `frontend/src/tests/e2e/ckbtc.spec.ts` which receives 20 BTC and checks that it arrives (minus fees).
5. Update images test to expect the ckBTC logo in the universe picker.
6. Update screenshots.

# Tests

pass

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -640,7 +640,7 @@
         "OPTIMIZER_VERSION": "0.3.6",
         "BINSTALL_VERSION": "1.3.0",
         "DIDC_VERSION": "2023-09-27",
-        "SNSDEMO_RELEASE": "release-2023-09-27",
+        "SNSDEMO_RELEASE": "release-2023-10-09",
```
Comment (lmuntaner) [inline dfx.json:643]: You might have a conflict after #3533 is merged. Please remember to use the one there, hehe.

Thanks!

File: frontend/src/tests/e2e/images.spec.ts [hunk 2]
```diff
@@ -17,14 +17,17 @@ const expectImagesLoaded = async ({ page, sources }) => {
...
       const images = Array.from(document.querySelectorAll("img"));
       if (images.length !== expectedImageCount) {
         return false;
       }
       // The browser might decide not to load images that are outside the
       // viewport.
-      images.forEach((img) => img.scrollIntoView());
+      for (const img of images) {
+        img.scrollIntoView();
+        await new Promise((resolve) => setTimeout(resolve, 0));
```
Comment (lmuntaner) [inline frontend/src/tests/e2e/images.spec.ts:29]: Why is this needed? Can you add a comment?

File: frontend/src/tests/e2e/ckbtc.spec.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  await appPo.getSelectUniverseListPo().goToCkbtcUniverse();
+
+  step("Check BTC balance");
+  await appPo
+    .getAccountsPo()
+    .getCkBTCAccountsPo()
+    .waitForWithdrawalAccountDone();
+
+  step("Get BTC");
+  await appPo.getIcpTokens(20);
```
Comment (lmuntaner) [inline frontend/src/tests/e2e/ckbtc.spec.ts:25]: We might have to rename this method, hehe.

File: frontend/src/tests/e2e/ckbtc.spec.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  step("Open ckBTC wallet");
+  await appPo
+    .getAccountsPo()
+    .getCkBTCAccountsPo()
+    .getMainAccountCardPo()
+    .click();
+
+  step("Refresh Balance");
+  await appPo.getWalletPo().getCkBTCWalletPo().clickRefreshBalance();
```
Comment (lmuntaner) [inline frontend/src/tests/e2e/ckbtc.spec.ts:35]: Doesn't it refresh automatically when you enter the page? Is this needed?

**Review by lmuntaner — APPROVED** [review]
Thanks! Only a few comments.

File: frontend/src/tests/e2e/ckbtc.spec.ts (hunk 4 above)
Comment (dskloetd) [reply frontend/src/tests/e2e/ckbtc.spec.ts:35]: Yes, we wait for it to finish on line 22 before we receive the BTC (on line 25). Then we click the "Refresh Balance" button to do it again after receiving the BTC.

File: frontend/src/tests/e2e/images.spec.ts (hunk 2 above)
Comment (dskloetd) [reply frontend/src/tests/e2e/images.spec.ts:32]: Done.
